File: quickstart/my_strategy.py
from model.game import Game
from model.order import Order
from model.unit_order import UnitOrder
from model.constants import Constants
from model.vec2 import Vec2
from model.action_order import ActionOrder
from typing import Optional
from debug_interface import DebugInterface


class MyStrategy:
    def __init__(self, constants: Constants):
        pass
    def get_order(self, game: Game, debug_interface: Optional[DebugInterface]) -> Order:
        orders = {}
        for unit in game.units:
            if unit.player_id != game.my_id:
                continue
            orders[unit.id] = UnitOrder(
                Vec2(-unit.position.x, -unit.position.y),
                Vec2(-unit.direction.y, unit.direction.x),
                ActionOrder.Aim(True))
        return Order(orders)
    def debug_update(self, displayed_tick: int, debug_interface: DebugInterface):
        pass
    def finish(self):
        pass

File: quickstart/my_strategy.rb
class MyStrategy
  def initialize(constants)
  end

  def get_order(game, debug_interface)
    orders = {}
    game.units.each do |unit|
      next if unit.player_id != game.my_id

      orders[unit.id] = Model::UnitOrder.new(
        Model::Vec2.new(-unit.position.x, -unit.position.y),
        Model::Vec2.new(-unit.direction.y, unit.direction.x),
        Model::ActionOrder::Aim.new(true)
      )
    end

    Model::Order.new(orders)
  end

  def debug_update(displayed_tick, debug_interface)
  end

  def finish()
  end
end
